src/components/AnnotationCreator.jsx

```diff
@@ -229,16 +301,49 @@ export default function AnnotationCreator({
               ),
             ];
             const theta = get(rect, 'theta', 0);
+            const newEncounterGuids = pending
+              ? []
+              : await createEncounter(
+                  sightingData,
+                  addEncounterToSighting,
+                );
+            if (addEncounterToSightingError)
+              setErrors(prevErrors => [
+                ...prevErrors,
+                addEncounterToSightingError,
+              ]);
+
+            const newEncounterGuid = get(newEncounterGuids, [0]);
+
             const newAnnotationId = await postAnnotation(
               assetId,
               IAClass,
               coords,
               viewpoint,
               theta,
+              newEncounterGuid,
             );
+            if (postAnnotationError)
+              setErrors(prevErrors => [
+                ...prevErrors,
+                postAnnotationError,
+              ]);
+
             if (newAnnotationId) {
+              if (newEncounterGuid) {
```

Comment:
simplify with `if (newAnnotationId && newEncounterGuid)`

Reply:
newAnnotationId gets created both if pending and if !pending.
However, addAnnotationsToSightingEncounter should only be run if !pending (and, indeed, more specifically only if there is a newEncounterGuid that can be used for said addAnnotationsToSightingEncounter call).

In other words, the two if statements above really ought to remain separate.